src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,2091 @@
+        Return:
+            hidden_state (`torch.Tensor` of shape `(batch_size, num_channels, sequence_length, d_model)`)
+
+            all_hidden_states (*optional*, returned when `output_hidden_states` is set to True, tuple of `torch.Tensor`
+            of shapes `(batch_size, num_channels, sequence_length, d_model)`)
+
+        """
+        all_hidden_states = []
+
+        for mod in self.layers:
+            hidden_state = mod(hidden_state)
+            if output_hidden_states:
+                all_hidden_states.append(hidden_state)
+        if output_hidden_states is False:
+            return hidden_state, None
+        return hidden_state, all_hidden_states
+
+
+class PatchTSTEncoderLayer(nn.Module):
+    """
+    PatchTST encoder layer
+    """
+
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__()
+
+        self.channel_attention = config.channel_attention
+
+        # Multi-Head attention
+        self.self_attn = PatchTSTAttention(
+            embed_dim=config.d_model,
+            num_heads=config.encoder_attention_heads,
+            dropout=config.attention_dropout,
+        )
+
+        # Add & Norm of the sublayer 1
+        self.dropout_path1 = nn.Dropout(config.dropout_path) if config.dropout_path > 0 else nn.Identity()
+        if "batch" in config.norm.lower():
+            self.norm_sublayer1 = nn.Sequential(
+                PatchTSTTranspose(1, 2), nn.BatchNorm1d(config.d_model), PatchTSTTranspose(1, 2)
+            )
+        else:
+            self.norm_sublayer1 = nn.LayerNorm(config.d_model)
+
+        # Add & Norm of the sublayer 2
+        if self.channel_attention:
+            self.dropout_path2 = nn.Dropout(config.dropout_path) if config.dropout_path > 0 else nn.Identity()
+            if "batch" in config.norm.lower():
+                self.norm_sublayer2 = nn.Sequential(
+                    PatchTSTTranspose(1, 2), nn.BatchNorm1d(config.d_model), PatchTSTTranspose(1, 2)
+                )
+            else:
+                self.norm_sublayer2 = nn.LayerNorm(config.d_model)
+
+        # Position-wise Feed-Forward
+        self.ff = nn.Sequential(
+            nn.Linear(config.d_model, config.encoder_ffn_dim, bias=config.bias),
+            ACT2CLS[config.activation_function](),
+            nn.Dropout(config.ff_dropout) if config.ff_dropout > 0 else nn.Identity(),
+            nn.Linear(config.encoder_ffn_dim, config.d_model, bias=config.bias),
+        )
+
+        # Add & Norm of sublayer 3
+        self.dropout_path3 = nn.Dropout(config.dropout_path) if config.dropout_path > 0 else nn.Identity()
+        if "batch" in config.norm.lower():
+            self.norm_sublayer3 = nn.Sequential(
+                PatchTSTTranspose(1, 2), nn.BatchNorm1d(config.d_model), PatchTSTTranspose(1, 2)
+            )
+        else:
+            self.norm_sublayer3 = nn.LayerNorm(config.d_model)
+
+        self.pre_norm = config.pre_norm
+
+    def forward(self, hidden_state: torch.Tensor):
```

Comment:
this class should support output_attentions just like `BertLayer` for instance

Reply:
Thank you! output_attentions flag is now available for all the classes.

Reply:
Yes, output_attentions flag is added on all the classes